crates/medea_macro/src/enum_delegate.rs

```diff
@@ -0,0 +1,100 @@
+//! `#[enum_delegate]` macro implementation.
+
+use std::iter;
+
+use proc_macro::TokenStream;
+use quote::quote;
+
+/// Generates the actual code for `#[enum_delegate]` macro.
+///
+/// # Generation algorithm
+/// 1. parse input enum;
+/// 2. check that enum is not empty;
+/// 3. add "{}" to the macro argument to get the full function declaration;
+/// 4. parse this full function declaration;
```

Comment:
Too verbose. The code here is quite explanatory itself. Only leaving main points has sense.